src/Prove.hs

```diff
@@ -119,9 +128,14 @@ mkContexts inv@(Invariant contract _) behv@(Behaviour method _ _ c1 (Interface _
   postStore <- Map.fromList <$> mapM (mkSymStorage method Post) locs
   env <- mkEnv contract method
 
+  calldata' <- case initBehv of
+                 Nothing -> pure $ calldata
+                 Just (Behaviour _ _ _ _ (Interface _ initDecls) _ _ _ _) -> do
+                   initCd <- Map.fromList <$> mapM (mkSymArg contract method) initDecls
+                   pure $ Map.unionWith (\l _ -> l) initCd calldata
```

Comment:
ummm, I think this is just normal union. Though I guess you want something like `Map.unionWith (error "TODO: duplicated calldata arguments")` for now or something

Reply:
oh didn't know about normal union haha :grinning: